pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -37,6 +38,10 @@ const (
 	inputParamContextDir = "CONTEXT_DIR"
 )
 
+var (
+	systemReservedParametersRegexp = regexp.MustCompile("BUILDER_IMAGE|DOCKERFILE|CONTEXT_DIR|(shp-.*)")
```

Comment:
Could this instead be expressed as a helper function that checks if the string is in a set of disallowed strings, and doesn't have the `shp-` prefix?

I find that regular expressions can be hard to read and understand, which can lead to bugs. If it can be implemented in reasonably brief code that's usually better in my experience:

```
var reserved := map[string]bool{
  "BUILDER_IMAGE": true,
  "DOCKERFILE":    true,
  "CONTEXT_DIR":   true,
}
func isReserved(n string) bool {
  return reserved[n] || strings.HasPrefix("shp-")
}
```

Reply:
yes, nice, done.